### ISSUE TITLE
about the view modularization

### ISSUE THREAD

**Issue by jonirrings** [jonirrings/react-stack#4]
about the pure view layer modularization, for now, we use only a component directory to hold all, it is not a good solution for modularization, cause it makes all level of view modules onto a flat plane. so I think another way, how about.
```
views
├─components
└─pages
```
any suggestion?

**Comment by jamesallain** [comment]
Yeah, that sounds good. I believe that's the recommended pattern. I like name "pages" better than "containers" too. 

I've been learning Typescript. It's really great. The only hard part is setting up the project so it transpiles easily. Also, started using [FeathersJS](https://feathersjs.com/). It's really cool. I've had trouble with OAuth2 and authentication. It makes it easy.

**Comment by jamesallain** [comment]
Also, [Expo](https://expo.io/) is AMAZING! Genymotion and Android Studio are pretty nasty in comparison. I might have my first React Native - Modern Relay app today :)

**Comment by jamesallain** [comment]
[ReactXP](https://microsoft.github.io/reactxp/)  looks pretty awesome too. I just stumbled on it. Using it with Graphql and Relay could be the Holy Grail of starter packages.

**Comment by jonirrings** [comment]
seems you are building react-native app, that is great.
for TypeScript with react, there are some grammer difference from the JS with react.
for instance, to import a module without default export.
TS
```ts
import * as React from 'react';
```
babel JS
```js
import React from 'react';
```
in fact, I think the TS style more formal and misleading.

**Comment by jamesallain** [comment]
Typescript is definitely more verbose. I think it has the potential to become the standard for app development. With Google and Microsoft, it definitely has the backing. I don't think Flow is going to make it.

ReactXP isn't ready, but it looks like it could be a winner.

**Comment by jonirrings** [comment]
yes, most IT companies in my place had adopt TS. seems it is accepted and examined by many companies.
